### ISSUE TITLE
intersect_edges does not appear to be entirely robust

### ISSUE THREAD

**Issue by Huite** [Deltares/numba_celltree#26]
This is fine:

```python
import numpy as np
import numba_celltree

triangles = np.fromfile("c:/tmp/triangles.bin", dtype=np.int64).reshape((-1, 3))
vertices = np.fromfile("c:/tmp/vertices.bin", dtype=np.float64).reshape((-1, 2))

tree = numba_celltree.CellTree2d(vertices, triangles, -1)

# %%

xmin = vertices[:, 0].min()
ymin = vertices[:, 1].min()
xmax = vertices[:, 0].max()
ymax = vertices[:, 1].max()
x = np.linspace(xmin, xmax, 1000)
y = np.linspace(ymin, ymax, 1000)
yy, xx = [a.ravel() for a in np.meshgrid(y, x, indexing="ij")]
points = np.column_stack((xx, yy))
edges = points.reshape((-1, 2, 2))

# %%

tree.intersect_edges(edges)
# %%
```

But testing a grid against its own edges either enters a infinite loop or something else:

```python
edges = numba_celltree.demo.edges(triangles, -1)
edge_coords = vertices[edges]
tree.intersect_edges(edge_coords)
```

Needs testing with a smaller example.

**Comment by Huite** [comment]
This is due to `dx`, `dy` not getting updated: https://github.com/Deltares/numba_celltree/blob/374a7a4a7606952d2487b0d96ec5c6603cfd286f/numba_celltree/algorithms/cohen_sutherland.py#L74

It can be triggered by this:

![Image](https://github.com/user-attachments/assets/92548190-d850-4987-ab69-c0e1f31c9035)

Where `b` is equal to (xmin, ymax).

Liang-Barsky is also not consistent here, since it returns `True` for intersection although the resulting edge is a single point.

**Comment by Huite** [comment]
Fixed by f47aa91c49af61657c8d8bb5d8aa60556e251518
